include/Beman/Optional26/optional.hpp

```diff
@@ -546,38 +538,56 @@ class optional {
     template <class F>
     constexpr auto transform(F&& f) & {
         using U = std::invoke_result_t<F, T&>;
+        static_assert(!std::is_array_v<U>);
+        static_assert(!std::is_same_v<U, in_place_t>);
+        static_assert(!std::is_same_v<U, nullopt_t>);
+        //        static_assert(std::is_object_v<U>); /// References now allowed
```

Comment:
reminder

Reply:
It's the comment on the end that's important. This has to make it back into the optional<T&> paper as it's a change I hadn't realized needed to be made to the base template.

Although I've been thinking about it and it might need to be more complicated. It needs to be, I think now, either a reference -- to any type -- or an object that can be stored in the optional. 
std::is_reference_v<U> || std::is_object_v<U>

At least disjunction in a static_assert isn't expensive like in concept constraints.

Reply:
Then add a TODO inside the comment.